chainsaw-demo/src/bin/chainsaw-demo-http/config.rs

```diff
@@ -0,0 +1,75 @@
+use std::net::{IpAddr, SocketAddr};
+
+use config::Config;
+use serde::Deserialize;
+
+#[derive(Deserialize)]
+pub struct Chainsaw {
+    pub http: HTTP,
+
+    pub log: Log,
+}
+
+#[derive(Deserialize)]
+pub struct HTTP {
+    #[serde(default = "default_http_address")]
+    pub address: IpAddr,
+    pub port: u16,
+}
+
+impl HTTP {
+    pub fn serve_addr(&self) -> SocketAddr {
+        SocketAddr::new(self.address, self.port)
+    }
+}
+
+#[derive(Deserialize)]
+pub struct Log {
+    #[serde(default)]
+    pub level: LogLevel,
+}
+
+#[derive(Clone, Copy, Debug, Deserialize, PartialEq)]
+#[serde(rename_all = "lowercase")]
+pub enum LogLevel {
+    Error,
+    Warn,
+    Info,
+    Debug,
+    Trace,
+}
+
+impl From<LogLevel> for tracing::Level {
+    fn from(level: LogLevel) -> Self {
+        match level {
+            LogLevel::Error => Self::ERROR,
+            LogLevel::Warn => Self::WARN,
+            LogLevel::Info => Self::INFO,
+            LogLevel::Debug => Self::DEBUG,
+            LogLevel::Trace => Self::TRACE,
+        }
+    }
+}
+
+impl Default for LogLevel {
+    fn default() -> Self {
+        Self::Info
+    }
+}
```

Comment:
As this is likely to be super common, I wondered about making a config module that could be shared/consumed instead of copy/pasted with this in.

Reply:
Agreed.